### PR TITLE
updated mauna-loa-co2 data

### PR DESCRIPTION
**Pull request by dragazo**
Updated the mauna loa co2 data file to most recent version. Apparently they changed the data file format, so I also update the loader. I was hoping to make it auto-download the data on startup, but if they keep changing the structure that's not gonna work. Oh well.

### CODE REVIEW

File: src/server/services/procedures/mauna-loa-co2-data/mauna-loa-co2-data.js [hunk 1]
```diff
@@ -7,24 +7,20 @@
...
-            .map(txt => parseFloat(txt));
-        return {date, avg, interpolated, trend};
+    return lines.filter(x => !x.startsWith('#')).map(line => {
+        const [,, date, interpolated, trend] = line.split(/\s+/).map(parseFloat);
+        return {date, interpolated, trend};
     });
 })();
 
 const MaunaLoaCO2Data = {};
-MaunaLoaCO2Data.serviceName = 'MaunaLoaCO2Data';
```
Comment (brollb) [inline src/server/services/procedures/mauna-loa-co2-data/mauna-loa-co2-data.js:27]: Why did you remove this line? Won't this change the name to "MaunaLoaCo2Data" (note the lowercase "o")?

File: src/server/services/procedures/mauna-loa-co2-data/mauna-loa-co2-data.js [hunk 2]
```diff
@@ -7,24 +7,20 @@
...
-
-    while (lines[0].startsWith('#')) {
-        lines.shift();
-    }
-    return lines.map(line => {
-        const [,, date, avg, interpolated, trend] = line.split(/\s+/)
-            .map(txt => parseFloat(txt));
-        return {date, avg, interpolated, trend};
+    return lines.filter(x => !x.startsWith('#')).map(line => {
+        const [,, date, interpolated, trend] = line.split(/\s+/).map(parseFloat);
```
Comment (brollb) [inline src/server/services/procedures/mauna-loa-co2-data/mauna-loa-co2-data.js:18]: It's not obvious to me that the "average" values in the new dataset are the same as the "interpolated" (as opposed to the "average" column) in the old dataset.

I think this is fine but wanted to make a note here in case there are any surprises after updating as this is a possible culprit.

**Review by brollb — CHANGES_REQUESTED** [review]
If you could address the question about the removal of `serviceName`, that would be great! I think  the other comment is fine but still wanted to make a note of it.

File: src/server/services/procedures/mauna-loa-co2-data/mauna-loa-co2-data.js (hunk 2 above)
Comment (dragazo) [reply src/server/services/procedures/mauna-loa-co2-data/mauna-loa-co2-data.js:18]: On line 49 of the data file it comments that missing months are interpolated. The 'average' column in the old data set occasionally had `-99.99` placeholders. So I think they just got rid of the 'average' column and replaced it with 'interpolated'.

File: src/server/services/procedures/mauna-loa-co2-data/mauna-loa-co2-data.js (hunk 1 above)
Comment (dragazo) [reply src/server/services/procedures/mauna-loa-co2-data/mauna-loa-co2-data.js]: Oh, yeah. Ok, I just fixed that. I must've been thinking NetsBlox got the name from the export object name when it does the jsdoc parsing, so it looked redundant, but it actually gets it from the folder/file.
